### ISSUE TITLE
s390x (IBM Z) support to pass ignition file to CoreOS guest

### ISSUE THREAD

**Issue by barthy1** [coreos/mantle#1045]
fw_cfg is used to share igntion file with CoreOS guest
https://github.com/coreos/mantle/blob/master/platform/qemu.go#L283
However fw_cfg parameter is not supported for s390x (IBM Z) and ppc64le (Power)

The proposal is to use [guestfish](http://libguestfs.org) for architectures, which do not support fw_cfg, like it is done for coreos-assembler https://github.com/coreos/coreos-assembler/blob/master/src/cmd-run#L239

You can also see example of the implementation and more details here https://github.com/dmacvicar/terraform-provider-libvirt/pull/629

**Comment by lucab** [comment]
Thanks for the report!

> fw_cfg parameter is not supported for s390x (IBM Z) and ppc64le (Power)

I don't think there is any technical limitation for not having support those architectures (but I may be wrong, as I'm not an expert in that area).
Do we have ticket to track adding support for those two? What's the current status? Perhaps we can increase the priority of those a bit, if we indeed plan to have consumers?

**Comment by bgilbert** [comment]
@lucab The QEMU developers don't want fw_cfg used in this way and aren't excited about adding a generic userdata mechanism.  They've suggested alternatives, several of which won't work because they require waiting indefinitely for udev.  https://github.com/coreos/ignition/issues/666 and https://github.com/coreos/ignition/issues/825 have some of the context.  One suggestion was that we pass userdata via a virtual PCI device, which might work but we haven't pursued it yet.

Our official guidance for passing userdata on s390x shouldn't be to use guestfish, so we'll still need a proper metadata mechanism.  It might be okay for kola to use guestfs in the short term, to unblock testing, but we shouldn't expect that to be the long-term approach.

**Comment by lucab** [comment]
Sorry, I was not aware of the additional discussion on `fw_cfg`. The backchannel discussion for `s390x.qemu` is already tracked at https://github.com/coreos/ignition/issues/825 then.

**Comment by barthy1** [comment]
@lucab So would it be ok for you to use guestfish in the short term to unblock testing activities?

**Comment by barthy1** [comment]
https://github.com/coreos/mantle/pull/1050
